pkg/controllers/clusterset/clustersetmapper/clustersetmapper.go

```diff
@@ -56,40 +58,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// put all managedClusterSets into queue if there is the managedCluster event
-	err = c.Watch(&source.Kind{Type: &clusterv1.ManagedCluster{}},
-		&handler.EnqueueRequestsFromMapFunc{
-			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
-				if _, ok := obj.Object.(*clusterv1.ManagedCluster); !ok {
-					// not a managedCluster, returning empty
-					klog.Error("managedCluster handler received non-managedCluster object")
-					return []reconcile.Request{}
-				}
-
-				managedClusterSets := &clusterv1alpha1.ManagedClusterSetList{}
-				err := mgr.GetClient().List(context.TODO(), managedClusterSets, &client.ListOptions{})
-				if err != nil {
-					klog.Errorf("failed to list managedClusterSet %v", err)
-				}
-
-				var requests []reconcile.Request
-				for _, managedClusterSet := range managedClusterSets.Items {
-					requests = append(requests, reconcile.Request{
-						NamespacedName: types.NamespacedName{
-							Name: managedClusterSet.Name,
-						},
-					})
-				}
-
-				klog.V(5).Infof("List managedClusterSet %+v", requests)
-				return requests
-			}),
-		})
-	if err != nil {
-		return nil
-	}
-
-	if err = c.Watch(&source.Kind{Type: &clusterv1alpha1.ManagedClusterSet{}},
```

Comment:
I think you still need to watch managedClusterSet, what if a managedClusterSet is  added